scripts/simplify_network.py

```diff
@@ -522,6 +523,151 @@ def consense(x):
+
+    if len(isl_empty) == 0:
+        return n
+
+    n.mremove("Bus", isl_empty)
+    n.determine_network_topology()
+
+    load_sum_final = n.loads_t.p_set.sum().sum()
+    generators_sum_final = n.generators.p_nom.sum()
+
+    logger.info(
+        f"Dropped {len(isl_empty)} buses. A resulted load discrepancy is {(100 * ((load_sum_final - load_sum_origin)/load_sum_origin)):2.1}% and {(100 * ((generators_sum_final - generators_sum_origin)/generators_sum_origin)):2.1}% for average load and generation capacity, respectivelly"
+    )
+
+    return n
+
+
+def drop_isolated_nodes(n, drop_e_threshold, n_periods_in_year=8760):
+    n.determine_network_topology()
+
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    loads_t_ann = n_periods_in_year * n.loads_t.p_set.mean(axis=0)
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    load_islands = loads_t_ann.loc[loads_t_ann.index.isin(i_islands)]
+    # drop_e_threshold, GWh
+    small_load_index = load_islands[load_islands <= (1e3 * drop_e_threshold)].index
+    i_islands_drop = i_islands[i_islands.isin(small_load_index)]
+
+    if (i_islands.empty) | (len(i_islands_drop) == 0):
+        return n
+
+    i_loads_drop = n.loads[n.loads.bus.isin(i_islands_drop)].index
+    i_generators_drop = n.generators[n.generators.bus.isin(i_islands_drop)].index
+
+    n.mremove("Bus", i_islands_drop)
+    n.mremove("Load", i_loads_drop)
+    n.mremove("Generator", i_generators_drop)
+
+    n.determine_network_topology()
+
+    load_sum_final = n.loads_t.p_set.sum().sum()
+    generators_sum_final = n.generators.p_nom.sum()
+
+    logger.info(
+        f"Dropped {len(i_islands)} buses. A resulted load discrepancy is {(100 * ((load_sum_final - load_sum_origin)/load_sum_origin)):2.1}% and {(100 * ((generators_sum_final - generators_sum_origin)/generators_sum_origin)):2.1}% for average load and generation capacity, respectivelly"
+    )
+
+    return n
+
+
+def merge_isolated_nodes(
+    n, merge_e_threshold=0.0, aggregation_strategies=dict(), n_periods_in_year=8760
+):
+    generators_sum_origin = n.generators.p_nom.sum()
+    load_sum_origin = n.loads_t.p_set.sum().sum()
+
+    n.determine_network_topology()
+
+    # duplicated sub-networks mean that there is at least one interconnection between buses
+    i_islands = n.buses[
+        (~n.buses.duplicated(subset=["sub_network"], keep=False))
+        & (n.buses.carrier == "AC")
+    ].index
+
+    # merge buses having sufficient attached load
+    loads_t_ann = n_periods_in_year * n.loads_t.p_set.mean(axis=0)
+    load_islands = loads_t_ann.loc[loads_t_ann.index.isin(i_islands)]
+
+    # merge_e_threshold, GWh
+    suffic_load_index = load_islands[load_islands >= (1e3 * merge_e_threshold)].index
```

Comment:
similarly to above, please move the 1e3 outside the function so to keep it without hard-coded values

Reply:
Removed